### PR TITLE
add nccl, cvmfs, cuda-mpi modules, update mpich and gpu modules

### PR DESCRIPTION
**Pull request by lastephey**
Adds a podman-hpc version of the current shifter NCCL ss11 module

WIP since it still needs to be fully tested on muller

### CODE REVIEW

**Comment by lastephey** [comment]
Ok should be ready for review @danfulton and @scanon. To work correctly requires https://github.com/NERSC/podman-hpc/pull/41

Should now match what is in https://gitlab.nersc.gov/nersc/csg/nersc-cle/-/merge_requests/3885

File: etc/modules.d/nccl.yaml
Comment (danfulton) [inline etc/modules.d/nccl.yaml:1]: Should we remove the `bind:` or make it `bind: []`?  I know the `hook_tool` will handle it, but not sure if it's technically valid yaml.

**Comment by danfulton** [comment]
Otherwise, LGTM.

**Review by danfulton — CHANGES_REQUESTED** [review]

File: etc/modules.d/nccl.yaml
Comment (danfulton) [reply etc/modules.d/nccl.yaml:1]: No strong opinion, I think python just parses it as `{"bind": None}`

Comment (lastephey) [reply etc/modules.d/nccl.yaml:1]: Looks like it's optional, so I'll just remove.

Comment (danfulton) [reply etc/modules.d/nccl.yaml:1]: I confirm the latest `hook_tool` PR works with both absent and `None` sections... that was a bug I encountered and fixed.  :)

**Comment by danfulton** [comment]
LGTM

**Comment by danfulton** [comment]
Oops, I meant to close my review not the PR... sorry.

**Review by danfulton — APPROVED** [review]

**Pull request merged**